src/main.js

```diff
@@ -1,6 +1,41 @@
 import Vue from 'vue'
+import VueRouter from 'vue-router'
 import App from './App.vue'
+import Home from './pages/Home/Home.vue'
+import Detail from './pages/Detail/Detail.vue'
+import store from './store';
+
+Vue.use(VueRouter) // tells Vue that VouRouter exists
+
+const routes = [
+  {
+    path: '/home',
+    component: Home
+  },
+  {
+    path: '/',
+    redirect: '/home'
+  },
+  {
+    path: '/listing/:id',
+    name: 'listing',
+    component: Detail
+  }
+]
+
+// create the router instance and pass the `routes` option
+export const router = new VueRouter({
+  mode: 'history',
+  routes
+})
```

Comment:
What do you think move this code to other file ? 
Because the routes control isn't responsibility of `main.js`.

Never forget about responsibility !!!

Reply:
Yes, I think it's necessary.
Like `router > index.js`, having the configuration and telling Vue that VueRouter exists.

Reply:
Done!